kiwmi/desktop/output.c

```diff
@@ -279,6 +280,19 @@ output_destroy_notify(struct wl_listener *listener, void *UNUSED(data))
 
     wl_list_remove(&output->events.destroy.listener_list);
 
+    int n_layers = sizeof(output->layers) / sizeof(output->layers[0]);
+    for (int i = 0; i < n_layers; i++) {
+        struct kiwmi_layer *layer;
+        struct kiwmi_layer *tmp;
+        wl_list_for_each_safe (layer, tmp, &output->layers[i], link) {
+            wlr_layer_surface_v1_destroy(layer->layer_surface);
+        }
+
+        wl_list_remove(&output->layers[i]);
+        // Layers try to rearrange on destroy, it must stay valid
+        wl_list_init(&output->layers[i]);
```

Comment:
Looking at this again, i think combining with your original solution or adding a `kiwmi_output.destroying` may be worth it. 🤷